add script-dump document (#143)
valkey-io/valkey-doc: commands/script-dump.md

Reviewer: The filename is wrong.

> Returns the scripts in the script cache.

Reviewer 2: ```suggestion
Returns the original source code of a script in the script cache.
```

> This command accepts one SHA1 digest and return a script if the script

Reviewer 3: ```suggestion
This command accepts one SHA1 digest and returns a script if the script
```

> are already defined or NOSCRIPT error if not inside the script cache.

Reviewer 3: ```suggestion
is already defined or NOSCRIPT error if not inside the script cache.
```

> This command accepts one SHA1 digest and return a script if the script
> are already defined or not inside the script cache.
> This can be useful to debug. In some scenarios, the business may not be able to find the
> previously used Lua script and only have a SHA signature.
> Or there are multiple identical evalsha's args in monitor/slowlog,
> and admin is not able to distinguish the script body.

Reviewer 2: ```suggestion
This command accepts a SHA1 digest and returns the original script's source code if the script is present in the script cache.
This command is intended primarily for debugging, to introspect into the contents of a script when the user does not have access to script anymore.
One example is when an admin only has access to a script SHA from monitor or slowlog, and is unable to determine the contents of the script.
```
Code convention says one sentence per line. I also tried to simplify the wording a bit, feel free to take or leave some of my suggestions.

Reviewer: English speakers: Is it "a SHA1" or "an SHA1"? Does it depend how you pronounce SHA1? [ess-aych-ay-1] vs [sha-1]

Author: @zuiderkwast SHA is Secure Hash Algorithm, so i think it is "a SHA1".

Author: @madolson @zuiderkwast thanks for your suggestions
